src/utils.ts

```diff
@@ -1,15 +1,54 @@
 import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
 import { VisibilitySelectionMap } from './constants';
 
-export const getNewSelectionMap = (
+export const getNewMapAfterTopLevelCheckboxClick = (
   agentName: string,
   currentVisibilityMap: VisibilitySelectionMap
 ): VisibilitySelectionMap => {
   const newMap: VisibilitySelectionMap = { ...currentVisibilityMap };
-  if (currentVisibilityMap[agentName].length === 0) {
+  if (currentVisibilityMap[agentName].includes(agentName)) {
+    newMap[agentName] = [];
+  } else {
     newMap[agentName] = [agentName];
+  }
+  return newMap;
+};
+
+export const getNewMapAfterDisplayStateClick = (
+  agentName: string,
+  displayStateName: string,
+  allDisplayStates: string[],
+  currentVisibilityMap: VisibilitySelectionMap
```

Comment:
`VisibilitySelectionMap` should really reference what it's mapping to and from as a type.

`AgentsToSelectedChildren` or `AgentsToVisibilitySelection`?

Reply:
Agreed, I think `AgentsToVisibilitySelection` is better.... Is this another scope question? I'm using the same `VisibilitySelectionMap` as the website. Would it be confusing to have the same data structure named two different ways in NBSV and website? Might want some buy in on a name change from @meganrm and update it in both places...